lib/src/rules/avoid_init_to_null.dart

```diff
@@ -88,8 +90,13 @@ class _Visitor extends SimpleAstVisitor<void> {
   @override
   void visitDefaultFormalParameter(DefaultFormalParameter node) {
     var declaredElement = node.declaredElement;
-    if (declaredElement == null) {
-      return;
+    if (declaredElement == null) return;
+    if (declaredElement is DefaultSuperFormalParameterElementImpl) {
+      var superConstructorParameter = declaredElement.superConstructorParameter;
+      if (superConstructorParameter is DefaultFieldFormalParameterElementImpl) {
+        var defaultValue = superConstructorParameter.defaultValueCode;
+        if (defaultValue != null) return;
```

Comment:
It looks like this is too broad. To confirm or deny, consider adding a test with the following code:
```
class A {
  String? a;
  A({this.a = null});
}
class B extends A {
  B({super.a = null}); // LINT
}
```

Reply:
I think the test case you maybe meant was:

```dart
class A {
  String? a;
  A({this.a = ''});
}
class B extends A {
  B({super.a = null});
}
```

and indeed, this is not working as written!

Reply:
(Deleted a mistyped example.)